core/domain/question_services.py

```diff
@@ -153,6 +153,63 @@ def get_question_by_id(question_id, strict=True):
         return None
 
 
+def get_question_skill_links_of_skill(skill_id):
+    """Returns a list of QuestionSkillLink domains of a particular skill ID.
+
+    Args:
+        skill_id: str. ID of the skill
+
+    Returns:
+        list(QuestionSkillLink)|None. The list of question skill link
+        domains that are linked to the skill ID. None if it doesnt exist.
+    """
+
+    question_skill_link_model_list = (
+        question_models.QuestionSkillLinkModel.get_question_skill_models_of_skill( #pylint: disable=line-too-long
+            skill_id)
+    )
+    question_skill_link_domains = [
+        question_domain.QuestionSkillLink(
+            question_skill_link_model.question_id,
+            question_skill_link_model.skill_id)
+        for question_skill_link_model in question_skill_link_model_list
+    ]
+
+    return question_skill_link_domains
+
+
+def _save_question_skill_link(new_skill_id, question_skill_model_id):
+    """Updates the new skill id for the question skill link and saves it.
+
+    Args:
+        new_skill_id: str. New ID for the questions that are merged.
```

Comment:
This is a skill ID but your docstring implies it's an ID for questions. Also, the ID is actually defined by the question ID and skill ID so it seems incorrect to change the skill ID of the model but not the model's datastore ID.

Probably better to do a delete + create? And let's also put a guard on the storage model to forbid put() operations (raise Exception if that's attempted, the only access should be via create()).

The other thing to think about is that we need to handle "move skill" and "add skill" differently. For one we need to remove the previous link object. For the other, which represents a question with multiple skills, we would still preserve the existing link object.